src/index.tsx

```diff
@@ -137,15 +139,25 @@ export default class InfiniteScroll extends Component<Props, State> {
     }
   }
 
-  UNSAFE_componentWillReceiveProps(props: Props) {
+  componentDidUpdate(prevProps: Props) {
     // do nothing when dataLength is unchanged
-    if (this.props.dataLength === props.dataLength) return;
+    if (this.props.dataLength === prevProps.dataLength) return;
 
     this.actionTriggered = false;
-    // update state when new data was sent in
-    this.setState({
-      showLoader: false,
-    });
+  }
+
+  static getDerivedStateFromProps(nextProps: Props, prevState: State) {
+    const dataLengthChanged = nextProps.dataLength !== prevState.prevDataLength;
+
+    // reset when data changes
+    if (dataLengthChanged) {
+      return {
+        showLoader: false,
+        pullToRefreshThresholdBreached: false,
```

Comment:
instead of writing all state variables here, wouldn’t 
``` { ... prevState,  prevDataLength: nextProps.dataLength } ``` a better thing to do? 
This way if other state variables gets added which are unrelated to `getDerivedStateFromProps`, no change would be required here.
Thoughts?